spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -0,0 +1,237 @@
+
+|`ft.<name>.retry.callsFailed.total`
+| Counter | None
+| The number of times the method was called and ultimately failed after retrying
+
+|`ft.<name>.retry.retries.total`
+| Counter | None
+| The total number of times the method was retried
+|===
+
+Note that the sum of `ft_<name>_retry_calls_succeeded_not_retried_total`, `ft_<name>_retry_calls_succeeded_retried_total` and `ft_<name>_retry_calls_failed_total` will give the total number of calls for which the Retry logic was run.
+
+=== Metrics added for `@Timeout`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft.<name>.timeout.executionDuration`
+| Histogram | Nanoseconds
+| Histogram of execution times for the method
+
+|`ft.<name>.timeout.callsTimedOut.total`
+| Counter | None
+| The number of times the method timed out
+
+|`ft.<name>.timeout.callsNotTimedOut.total`
+| Counter | None
+| The number of times the method completed without timing out
+|===
+
+Note that the sum of `ft.<name>.timeout.callsTimedOut.total` and `ft.<name>.timeout.callsNotTimedOut.total` will give the total number of calls for which the Timeout logic was run. This may be larger than the total number of invocations of the method it's also annotated with `@Retry` because the Timeout logic is applied to each Retry attempt, not to the whole method invocation time.
+
+=== Metrics added for `@CircuitBreaker`
+
+[cols="8,3,4,9"]
+|===
+| Name | Type | Unit | Description
+
+|`ft.<name>.circuitbreaker.callsSucceeded.total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that returned successfully
+
+|`ft.<name>.circuitbreaker.callsFailed.total`
+| Counter | None
+| Number of calls allowed to run by the circuit breaker that then failed
+
+|`ft.<name>.circuitbreaker.callsPrevented.total`
+| Counter | None
+| Number of calls prevented from running by an open circuit breaker
+
+|`ft.<name>.circuitbreaker.timeOpen.total`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in open state
+
+|`ft.<name>.circuitbreaker.timeHalfOpen.total`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in half-open state
+
+|`ft.<name>.circuitbreaker.timeClosed.total`
+| Gauge<Long> | Nanoseconds
+| Amount of time the circuit breaker has spent in closed state
+
+|`ft.<name>.circuitbreaker.opened.total`
+| Counter | None
+| Number of times the circuit breaker has moved from closed state to open state
+|===
+
+Note that the sum of `ft.<name>.circuitbreaker.callsSucceeded.total`, `ft.<name>.circuitbreaker.callsFailed.total` and `ft.<name>.circuitbreaker.callsPrevented.total` will give the total number of calls for which the Bulkhead logic was run.
+
+Similarly, the sum of `ft.<name>.circuitbreaker.timeOpen.total`, `ft.<name>.circuitbreaker.timeHalfOpen.total` and `ft.<name>.circuitbreaker.timeClosed.total` will give the total time that the circuit breaker has been active for.
```

Comment:
I would say `ft.<name>.circuitbreaker.open.total` and `ft.<name>.circuitbreaker.halfOpen.total`

Reply:
Done